app/src/main/java/net/aiscope/gdd_app/ui/metadata/MetadataActivity.kt

```diff
@@ -34,37 +33,46 @@ class MetadataActivity : AppCompatActivity(), MetadataView, CaptureFlow {
     private val imagesAdapter =
         SampleImagesAdapter(lifecycleScope, this::onAddImageClicked, this::onImageClicked)
 
+    private lateinit var binding: ActivityMetadataBinding
+
     override fun onCreate(savedInstanceState: Bundle?) {
         AndroidInjection.inject(this)
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_metadata)
-
-        setSupportActionBar(toolbar)
-        attachCaptureFlowToolbar(toolbar)
 
-        metadata_blood_sample_images.apply {
-            setHasFixedSize(true)
-            layoutManager =
-                LinearLayoutManager(this@MetadataActivity, LinearLayoutManager.HORIZONTAL, false)
-            adapter = imagesAdapter
-        }
-
-        lifecycleScope.launch {
-            presenter.showScreen()
-        }
+        binding = ActivityMetadataBinding.inflate(layoutInflater)
+        with(binding) {
+            setContentView(root)
```

Comment:
This `root` feels a bit magic to me. But maybe that is because I am unfamiliar with android development.

Is there any way to refer to this that makes it more clear what exactly it is?

Reply:
Not really, `root` is how the field is called in the parent `ViewBinding` class that comes with the AndroidX library. But I would say that `root` is quite a common way to refer to the top-most view in a layout, so I don't think that this should be confusing for a regular Android developer.

Reply:
OK fair enough